tests/testthat/setup-testing_depth.R

```diff
@@ -0,0 +1,56 @@
+#' Returns testing depth set by an environmental variable.
+#'
+#' @details Looks for the option `TESTING_DEPTH` first, if not set,
+#' takes the system environmental variable `TESTING_DEPTH`. If neither
+#' is set, then returns 3 by default. If the value of `TESTING_DEPTH`
+#' is not a scalar numeric, then returns 3.
+#'
+#' @return `numeric(1)` the testing depth.
+#'
+testing_depth <- function() {
+  testing_depth <- getOption("TESTING_DEPTH")
```

Comment:
what about exposing the depth to be a parameter? so that you can set it during the function call OR with a session option.
```suggestion
testing_depth <- function(testing_depth = getOption("TESTING_DEPTH")) {

```

Reply:
I think having no parameter sends a stronger signal that the only way to influence the TEST DEPTH is to set it using options or env variable. Else we give the reader an additional task to figure out what is happening in the function call. So, I prefer not having a parameter to this function. Essentially it is like a read function to get the depth.

Reply:
Or is it just for getting the depth, and not setting? Then maybe rename to `get_testing_depth()`

Reply:
Yeah, let's have no parameter but let's change the name to `get_` as I spent some time figuring out if the purpose is to set (based on an ENVIRONEMNT VARIABLE / option) or to get the depth.

Reply:
Thanks! changed the name.